tests/communication/CMakeLists.txt

```diff
@@ -51,13 +82,43 @@ target_include_directories(app PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
 # Target
 # ######################################################################################################################
 
+# Enable AESNI for posix 32bit
+target_compile_options(mbedTLS INTERFACE -mpclmul -msse2 -maes)
+
+set(aoscore_source_dir "${CMAKE_CURRENT_SOURCE_DIR}/../../../aos_core_lib_cpp")
+
+target_compile_definitions(app PRIVATE CERT_DIR="${CERTDIR}")
+
 target_sources(
     app
     PRIVATE ../../src/communication/communication.cpp
             ../../src/communication/cmclient.cpp
+            ../../src/communication/tlschannel.cpp
+            ../../src/rootca/rootca.S
+            ${aoscore_source_dir}/src/common/crypto/mbedtls/driverwrapper.cpp
             ../../src/communication/iamserver.cpp
             ../../src/utils/checksum.cpp
             src/cmclient.cpp
             src/iamserver.cpp
             src/utils.cpp
+            src/tlschannel.cpp
+)
+
+set_source_files_properties(
+    ${ZEPHYR_MBEDTLS_MODULE_DIR}/library/psa_crypto_driver_wrappers_no_static.c
+    ${ZEPHYR_MBEDTLS_MODULE_DIR}/library/psa_crypto.c TARGET_DIRECTORY mbedTLS PROPERTIES HEADER_FILE_ONLY ON
 )
+
+target_sources(
+    app PRIVATE ${aoscore_source_dir}/src/common/crypto/mbedtls/drivers/psa_crypto_driver_wrappers_no_static.c
+                ${aoscore_source_dir}/src/common/crypto/mbedtls/drivers/psa_crypto.c
+)
+
+message(STATUS "AOS CORE SOURCE DIR: ${aoscore_source_dir}")
+
+set(CUSTOM_MBEDTLS_CONFIG_FILE ${aoscore_source_dir}/src/common/crypto/mbedtls/drivers/psa_crypto_driver_wrappers.h)
+
+configure_file(${CUSTOM_MBEDTLS_CONFIG_FILE} ${ZEPHYR_MBEDTLS_MODULE_DIR}/library/psa_crypto_driver_wrappers.h COPYONLY)
```

Comment:
Could we have the same mbedtls configuration as in the main cmake?

Reply:
removed